dataproc/src/main/java/Quickstart.java

```diff
@@ -0,0 +1,166 @@
+              .setMasterConfig(masterConfig)
+              .setWorkerConfig(workerConfig)
+              .build();
+      // Create the cluster object with the desired cluster config.
+      Cluster cluster =
+          Cluster.newBuilder().setClusterName(clusterName).setConfig(clusterConfig).build();
+
+      // Create the Cloud Dataproc cluster.
+      OperationFuture<Cluster, ClusterOperationMetadata> createClusterAsyncRequest =
+          clusterControllerClient.createClusterAsync(projectId, region, cluster);
+      Cluster response = createClusterAsyncRequest.get();
+      System.out.printf("Cluster created successfully: %s", response.getClusterName());
+
+      // Configure the settings for our job.
+      JobPlacement jobPlacement = JobPlacement.newBuilder().setClusterName(clusterName).build();
+      PySparkJob pySparkJob = PySparkJob.newBuilder().setMainPythonFileUri(jobFilePath).build();
+      Job job = Job.newBuilder().setPlacement(jobPlacement).setPysparkJob(pySparkJob).build();
+
+      // Submit an asynchronous request to execute the job.
+      Job request = jobControllerClient.submitJob(projectId, region, job);
+      String jobId = request.getReference().getJobId();
+      System.out.println(String.format("Submitted job \"%s\"", jobId));
+
+      // Wait for the job to finish.
+      CompletableFuture<Job> finishedJobFuture =
+          CompletableFuture.supplyAsync(() ->
+           waitForJobCompletion(jobControllerClient, projectId, region, jobId));
+      int timeout = 10;
+      try {
+        Job jobInfo = finishedJobFuture.get(timeout, TimeUnit.MINUTES);
+        System.out.printf("Job %s finished successfully.", jobId);
+
+        // Cloud Dataproc job output gets saved to a GCS bucket allocated to it.
+        Cluster clusterInfo = clusterControllerClient.getCluster(projectId, region, clusterName);
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        Blob blob =
+            storage.get(
+                clusterInfo.getConfig().getConfigBucket(),
+                String.format(
+                    "google-cloud-dataproc-metainfo/%s/jobs/%s/driveroutput.000000000",
+                    clusterInfo.getClusterUuid(), jobId));
+        System.out.println(
+            String.format(
+                "Job \"%s\" finished with state %s:\n%s",
+                jobId, jobInfo.getStatus().getState(), new String(blob.getContent())));
+      } catch (TimeoutException e) {
+        System.err.println(
+            String.format("Job timed out after %d minutes: %s", timeout, e.getMessage()));
+      }
+
+      // Delete the cluster.
+      OperationFuture<Empty, ClusterOperationMetadata> deleteClusterAsyncRequest =
+          clusterControllerClient.deleteClusterAsync(projectId, region, clusterName);
+      deleteClusterAsyncRequest.get();
+      System.out.println(String.format("Cluster \"%s\" successfully deleted.", clusterName));
+
+    } catch (ExecutionException e) {
```

Comment:
If `ExecutionException` occurs from interacting with the API (like it's thrown if you don't have permission or if your request is misconfigured), keep it and clarify that it's an error from interacting with the service with the `println`. 

If it's not from the API, please let it bubble up (and be consistent with the other samples - it's bubbled up in a few other places).

Reply:
Ahh yes, I see what you're saying. Errors caused by our API bubble up to be an Execution Exception, so I think it should definitely be caught. I'll add it the other sample.